libs/providers/go-feature-flag/src/lib/go-feature-flag-provider.ts

```diff
@@ -218,99 +171,51 @@ export class GoFeatureFlagProvider implements Provider {
   async resolveEvaluationGoFeatureFlagProxy<T>(
     flagKey: string,
     defaultValue: T,
-    user: GoFeatureFlagUser,
+    evaluationContext: EvaluationContext,
     expectedType: string,
   ): Promise<ResolutionDetails<T>> {
-    // Check if the provider is ready to serve
-    if (this._status === ProviderStatus.NOT_READY) {
-      return {
-        value: defaultValue,
-        reason: StandardResolutionReasons.ERROR,
-        errorCode: ErrorCode.PROVIDER_NOT_READY,
-        errorMessage: `Provider in a status that does not allow to serve flag: ${this.status}`,
-      };
-    }
-
-    const cacheKey = `${flagKey}-${hash(user)}`;
-    // check if flag is available in the cache
-    if (this.cache !== undefined) {
-      const cacheValue = this.cache.get(cacheKey);
+    if (this._cacheController?.get(flagKey, evaluationContext) !== undefined) {
+      const cacheValue = this._cacheController.get(flagKey, evaluationContext);
```

Comment:
Nit, but maybe you could save the result of `_cacheController?.get` in line 177 already and only check the value, then you can remove the following undefined check on the resolution value.